evidence/pool.go

```diff
@@ -55,10 +63,16 @@ func NewPool(evidenceDB dbm.DB, stateDB sm.Store, blockStore BlockStore) (*Pool,
 		logger:        log.NewNopLogger(),
 		evidenceStore: evidenceDB,
 		evidenceList:  clist.New(),
+		evidenceSize:  0,
+		pruningHeight: state.LastBlockHeight,
+		pruningTime:   state.LastBlockTime,
 	}
 
-	// if pending evidence already in db, in event of prior failure, then load it back to the evidenceList
-	evList := pool.AllPendingEvidence()
+	// if pending evidence already in db, in event of prior failure, then check for expiration,
+	// update the size and load it back to the evidenceList
+	pool.removeExpiredPendingEvidence()
+	evList := pool.allPendingEvidence()
```

Comment:
if allPendingEvidence errors and returns an empty list, what could be the worst consequences?

Reply:
Yeah good point. I guess you mean it errors even although it has evidence. I guess then if it comes to removing the evidence it will try to decrement the evidenceSize which is already 0

Reply:
I will just return the error here so if there is a failure when loading all the evidence it will return the error in `NewPool`